### PR TITLE
[Feature store] Add list features API + label support for features/entities

### PR DESCRIPTION
**Pull request by theSaarco**
Create a new API to query features (as opposed to feature-sets).
In addition, the following main items:
1. Support labels in feature-set features and entities
2. A more extensive set of tests for the feature-sets functionality
3. Some small fixes to feature-set logic

The following is known issue but deferred from this PR:
- When querying and using multiple label predicates, the SQL query generated is not correct and doesn't return values (not just for feature-sets/features)

### CODE REVIEW

File: mlrun/api/db/sqldb/db.py [hunk 1]
```diff
@@ -1231,20 +1321,40 @@ def _add_utc_timezone(obj, attribute_name):
     def _transform_feature_set_model_to_schema(
         feature_set_record: FeatureSet, tag=None,
     ) -> schemas.FeatureSet:
+        def _transform_label_list_to_dict(label_list):
+            return {label.name: label.value for label in label_list}
```
Comment (Hedingber) [inline mlrun/api/db/sqldb/db.py:1325]: Sounds like something that will be useful for other resources as well, you can move to outer scope

File: mlrun/api/schemas/feature_store.py [hunk 2]
```diff
@@ -56,3 +88,20 @@ class Config:
...
 class FeatureSetsOutput(BaseModel):
     feature_sets: List[FeatureSet]
+
+
+class FeatureSetDigestOutput(BaseModel):
+    name: str
+    tag: Optional[str]
+    labels: Optional[dict]
+    uid: Optional[str]
+    entities: List[Feature]
```
Comment (Hedingber) [inline mlrun/api/schemas/feature_store.py:98]: Let's reuse already existing schemas if possible, I suggest to simply use the regular `metadata: FeatureSetMetadata` here, plus I think I would make it `spec: FeatureSetDigestSpec`

File: mlrun/api/schemas/feature_store.py [hunk 3]
```diff
@@ -7,9 +7,13 @@
...
     value_type: str
+    labels: Optional[dict]
 
-    class Config:
-        orm_mode = True
+
+class Entity(BaseModel):
+    name: str
+    value_type: str
+    labels: Optional[dict]
```
Comment (Hedingber) [inline mlrun/api/schemas/feature_store.py:16]: I wonder whether we should make `Entity` and `Feature` to also have `metadata` and `spec` (currently no `status` I guess) WDYT ?
@yaronha ?

**Review by Hedingber — CHANGES_REQUESTED** [review]
Very nice!
Great test coverage
I would suggest to add a test for a more "extreme" usage of `list_features`, e.g. a scenario where you have multiple feature sets, and each feature set has multiple tags, each tags with different features (not a must in this PR)

File: mlrun/api/schemas/feature_store.py (hunk 3 above)
Comment (yaronha) [reply mlrun/api/schemas/feature_store.py:16]: @Hedingber no need, they are under featureset

File: mlrun/api/db/sqldb/db.py (hunk 1 above)
Comment (theSaarco) [reply mlrun/api/db/sqldb/db.py:1325]: ok. I moved it to the helpers.py file.

File: mlrun/api/schemas/feature_store.py (hunk 2 above)
Comment (theSaarco) [reply mlrun/api/schemas/feature_store.py:98]: Per our offline discussion - will make the changes.

**Review by Hedingber — APPROVED** [review]

**Pull request merged**